### ISSUE TITLE
[BUG] name saved in alias

### ISSUE THREAD

**Issue by Felecarpp** [evennia/evennia#3660]
#### Describe the bug
I would like to change the name of some of my objects. I use the `get_display_name` method for this. My implementation does some checks with `self.tags.has` to choose the right template to render the name of my object. 
But I observed that the old names that are no longer valid are saved as aliases. That is disturbing to me because the object respond to user keywords that no longer describe the object. 

#### To Reproduce
1. `create closed box`
2. `inventory`
3. `name closed box=opened box`
4. See error : `alias opened box`
5. See an other error : `search closed box`

#### Expected behavior
I would like aliases are created only when I manually register them.

#### Environment, Evennia version, OS etc
```
> about
Evennia MU* development system

Evennia version: 4.4.1
OS: posix
Python: 3.12.7
Twisted: 23.10.0
Django: 4.2.16
```

#### Additional context
Perhaps I should not dynamically change the name of an object. That would be disturbing to me because I need my objects to change state and to make it easily visible.

**Comment by Griatch** [comment]
I couldn't reproduce this at first, but I see now what is happening - the assignment of numbered aliases happen on-demand (as you say, using the inventory/look commands for example). 

You are 'tricking' the system here by doing an `inventory` command after creation of the box, but *not* after you rename it. If you do, the box will correctly refresh its aliases to the closed box. 

```
name closed box = opened box
py self.search("box").aliases.all()
['a closed box', 'one closed box']
inventory
py self.search("box").aliases.all()
['an opened box', 'one opened box']
```

This is actually working as intended; since numbered names are mainly used in visualization, they are calculated on-demand when that visualization is needed. But I can see it's confusing when trying to get this data programmatically.

**Comment by Felecarpp** [comment]
Thank you. How can I programmatically update the aliases ?

**Comment by Griatch** [comment]
@Felecarpp count-infinity made a PR to address this issue in https://github.com/evennia/evennia/commit/3ba4c6dbb13afe8b26bc7ec591bd397f2b3b4dc5 - their change basically has the `name` command forcibly clear the aliases before updating. Will be available in Evennia 4.5 :)
